### PR TITLE
End mentoring session

### PR DESCRIPTION
**Pull request by kntsoriano**
https://github.com/exercism/v3-project-management/issues/116

## Discussion
I've just guessed what the endpoints are and what they will be responding with. @iHiD can you add commits to this so we can get this working end to end?

## Questions
1. Based on the mock ups, there is a bug:
  - mentor favorites student
  - mentor clicks change preferences
  - mentor clicks to not mentor student again
  - Bug: student remains one of mentor's favorites

2. Since we also have the change preferences button, should we add a screen to "unfavorite" the student?

### CODE REVIEW

**Comment by iHiD** [comment]
@kntsoriano Functionally, this is all great. Nice work.

---

I've added some work to do the following:
- Renaming end-session to finish-discussion (I don't like having methods called `def end` - it fucks with my head). I did have it as `completed?`  but I've changed it now everywhere from to `finished?` 🙂 
- All the endpoints are implemented and work now (I've tested it all)
- I've styled the confirmation modal

I still have to do:
- Style the other bit
- Do a proper command for the finishing the discussion bit (I've made it functional but not done it properly yet).

Note: I can't get any JS tests to work. I've pinged you on Slack about this. So I can't test that bit.

---

The main problem we have is this:
https://slack-files.com/T1PRL3XMX-F01KX4H8138-9438991c58

I think I wasn't clear enough on this issue description for this. **Sorry.**

Rather than the modal being multiple part modal, it's an initial single yes/no modal (which you have). Then it shows the rest as a component in the bottom of the discussion, rather than on a modal. This should be at the bottom of the discussion whenever a discussion is `finished?`

Basically I think that everything from the confirmation button need move from being inside a modal to being part of the Discussion component, and then the discussion component needs to watch for `finished?` and render that thing (and scroll to it) if it changes. I don't think that's too much work, but sorry for not being clear on this before.

I was going to take a stab at doing this, but I think it'll be better for you to do it.

---

Also, when the discussion is finished, the "End discussion" button should disappear on the discussion page.

**Comment by iHiD** [comment]
@kntsoriano This is looking really good now. I've added a couple of commits.

There are four things blocking now: 
1. Implement [this TODO].(https://github.com/exercism/website/pull/475/commits/7d022e97b9208d0e25ca81ef72606aeae1afabf9#diff-7b9c0a44f81a8ca31930606f039b1698356ed90909635dd287f94aa7330107adR201)
1. Implement [this (similar) TODO](https://github.com/exercism/website/pull/475/files#diff-76fa47521c7ab5b3f6ade8e5912300254b4778efbb4bd4fd349b6c6330c9ec98R45-R48) - I'll then do the styling separately once there's a pivot there. This should change as soon as the discussion is finished. The screenshot below shows how it looks. It's still the same button/functionality - just a different style.
1. When the page is loading (ie not part of flow) the wizard should start in the final state
1. Fixing my [React here](https://github.com/exercism/website/pull/475/commits/3c0be754608ce190199f845037a6bec7805cb6d8#diff-83ee86da550936f236f1b3f51f03c9e941883c764c41b62bece34825b009e29bR16-R22) if it's bad 🙂 

---

Screenshot 2021-02-08 at 15 23 44

**Review by iHiD — APPROVED** [review]

**Comment by kntsoriano** [comment]
What do you mean by "When the page is loading (ie not part of flow) the wizard should start in the final state"?

I've done the TODOs listed.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by iHiD** [comment]
@kntsoriano
At the moment, every time you go the page it starts you back at the beginning at the wizard. The only time it should show the beginning of the wizard is the first time this appears. The other times it should go to the end.

ie. it should be like this:

- Mentor works on Solution
- Mentor clicks "End discussion"
- Wizard appears with the first question _(Happening correctly now)_

...

- Mentor returns back to the page later (or refreshes)
- Wizard shows the final state, with "change preferences". _(Currently incorrectly showing first page of wizard)_

**Comment by kntsoriano** [comment]
@iHiD Cool. I've done the needed changes.

**Comment by iHiD** [comment]
@kntsoriano Thanks. The tests are failing on this one still.
